### PR TITLE
Exclude AppKit extensions to be compiled for macCatalyst

### PR DESCRIPTION
**Pull request by MaxHaertwig**
## Checklist

- [/] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [/] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/SwifterSwift/AppKit/NSColorExtensions.swift [hunk 1]
```diff
@@ -6,7 +6,7 @@
 //  Copyright © 2019 SwifterSwift
 //
 
-#if canImport(AppKit)
+#if canImport(AppKit) && !targetEnvironment(macCatalyst)
```
Comment (omaralbeik) [inline Sources/SwifterSwift/AppKit/NSColorExtensions.swift]: Why don't we use `#if os(macOS)` instead here?

Comment (guykogus) [reply Sources/SwifterSwift/AppKit/NSColorExtensions.swift]: I think both are correct, technically. We use this `#if` elsewhere in the project, so this usage is in line with the others, and we should change it everywhere else if we use `#if os(macOS)` here.
If we're thinking about future-proofing, unfortunately it's impossible to know what will happen. Will `AppKit` suddenly be available on another platform that doesn't use `NSColor`? If so, we'll need to update all of these `#if`s. If the new platform does use `NSColor` but we've used `#if os(macOS)` then none of these extensions will be available on the new platform (and we won't know about it, most likely).

File: CHANGELOG.md [hunk 2]
```diff
@@ -47,6 +47,8 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
 
 ### Fixed
```
Comment (guykogus) [inline CHANGELOG.md]: Remove this empty line

File: Sources/SwifterSwift/AppKit/NSColorExtensions.swift (hunk 1 above)
Comment (omaralbeik) [reply Sources/SwifterSwift/AppKit/NSColorExtensions.swift]: Apple ¯\\_(ツ)_/¯

Comment (MaxHaertwig) [reply Sources/SwifterSwift/AppKit/NSColorExtensions.swift]: We also use `#if canImport(AppKit) && !targetEnvironment(macCatalyst)` in `ColorExtensions.swift` and other files.

Comment (guykogus) [reply Sources/SwifterSwift/AppKit/NSColorExtensions.swift]: @MaxHaertwig we already have :)

**Pull request merged**